tensorflow/core/kernels/data/prefetch_dataset_op.cc

```diff
@@ -144,6 +139,14 @@ class PrefetchDatasetOp::Dataset : public DatasetBase {
     }
 
     Status Initialize(IteratorContext* ctx) override {
+      TF_RETURN_IF_ERROR(
+          ConnectCancellationManagers(ctx->cancellation_manager(),
+                                      &cancellation_manager_, &deregister_fn_));
+      CancellationToken token = cancellation_manager_.get_cancellation_token();
```

Comment:
why are you registering two callbacks? (look at what is `ConnectCancellationManager` doing)

Reply:
I misunderstood the logic that the cancellation should happen in the destructor instead of the callback. Got it now with your other comment. Thanks, @jsimsa!